scripts/update_sales_values.ts

```diff
@@ -0,0 +1,187 @@
+        });
+        currencyAmount = transaction.value;
+      } else {
+        const currencyLogs = transactionReceipt.logs.filter(
+          (log) => log.address.toLowerCase() === currency.toLowerCase(),
+        );
+        const parsedCurrencyLogs = parseEventLogs({
+          abi: erc20Abi,
+          logs: currencyLogs,
+        });
+        const transferLog = parsedCurrencyLogs.find(
+          (log) => log.eventName === "Transfer",
+        );
+        currencyAmount = transferLog?.args?.value ?? 0n;
+      }
+
+      const exchangeLogs = transactionReceipt.logs.filter(
+        (log) =>
+          log.address.toLowerCase() ===
+          addresses?.HypercertExchange?.toLowerCase(),
+      );
+
+      const parsedExchangeLog = parseEventLogs({
+        abi: HypercertExchangeAbi,
+        logs: exchangeLogs,
+        // @ts-expect-error eventName is missing in the type
+      }).find((log) => log.eventName === "TakerBid");
+
+      // @ts-expect-error args is missing in the type
+      const fee_amounts = parsedExchangeLog?.args?.feeAmounts;
+      // @ts-expect-error args is missing in the type
+      const fee_recipients = parsedExchangeLog?.args?.feeRecipients;
+
+      results.push(
+        TakerBid.parse({
+          amounts: sale.amounts.map((amount) => BigInt(amount)),
+          seller: getAddress(sale.seller),
+          buyer: getAddress(sale.buyer),
+          currency: getAddress(sale.currency),
+          collection: getAddress(sale.collection),
+          item_ids: sale.item_ids.map((item_id) => BigInt(item_id)),
+          strategy_id: BigInt(sale.strategy_id),
+          hypercert_id: hypercert_id,
+          transaction_hash: sale.transaction_hash,
+          currency_amount: currencyAmount,
+          fee_amounts: fee_amounts,
+          fee_recipients: fee_recipients,
+        }),
+      );
+    } catch (e) {
+      console.log("Error parsing transaction", JSON.stringify(sale, null, 2));
+      console.log(e);
+      continue;
+    }
+  }
+
+  // Combine parsed results with original sales data by matching transaction hashes
+  const rowsToUpsert = results.map((result) => {
+    const originalSale = sales.find(
+      (sale) => sale.transaction_hash === result.transaction_hash,
+    );
+    if (!originalSale) {
+      throw new Error(
+        `Could not find original sale for transaction ${result.transaction_hash}`,
+      );
+    }
+    return {
+      ...originalSale,
+      ...result,
+      strategy_id: result.strategy_id.toString(),
+      item_ids: result.item_ids.map((id) => id.toString()),
+      amounts: result.amounts.map((amount) => amount.toString()),
+      currency_amount: result.currency_amount.toString(),
+      fee_amounts: result.fee_amounts.map((amount) => amount.toString()),
+    };
+  });
+
+  console.log(rowsToUpsert);
+
+  // Upsert rows
+  console.log("Upserting rows");
+  const res = await supabase.from("sales").upsert(rowsToUpsert);
+  console.log(res);
```

Comment:
Can you log an overview, so we can validate the changes made by spot checking the output?

Reply:
Can you explain what you would expect this overview to look like? It's already logging the updated rows above. Do we want a before/after? That's what I've added now, not sure what else to add.

Reply:
Yeah a before-after, sorry I was unclear. But then we can first roll out to staging and double check.